### PR TITLE
Graphql update sample metadata mutation

### LINKED ISSUE #381: GraphQL: updateSampleMetadata mutation
## Summary

As a User I would like to be able to update the Metadata for a Sample via the GraphQL.

## Acceptance Criteria
* Create a new `updateSampleMetadata` mutation the adhere's to the following criteria
  * Accepts the following arguments:
    * `sample_id` which is the Node ID of the project e.g. `gid://Sample/1`
    * `metadata` which is a JsonType
  * Returns the following fields:
    * `errors` an array of strings of any errors
    * `status` a JsonType which is the response of the `Samples::Metadata::UpdateService`
    * `sample` the updated sample after the metadata has been updated
  * This uses the existing `Samples::Metadata::UpdateService`

### PR DESCRIPTION
**Pull request by ksierks**
## What does this PR do and why?
Adding a UpdateSampleMetadata mutation to the GraphQL API which allows a user to update sample metadata.
Fixes #381.

## Screenshots or screen recordings
N/A

## How to set up and validate locally
1. Login as `admin@email.com`.
2. Navigate to `http://localhost:3000/graphiql` to open the graphiql interface .
3. Execute a mutation to update metadata for a Sample.
```graphql
{
  updateSampleMetadata(input: {sampleId: $sample_id, metadata: $metadata}) {
    sample{
      id,
      name,
      description,
      metadata
    },
    status,
    errors
  }
}
```
4. Verify that you see all the metadata for the sample was updated.

## PR acceptance checklist
This checklist encourages us to confirm any changes have been analyzed to reduce risks in quality, performance, reliability, security, and maintainability.

- [ ] I have evaluated the [PR acceptance checklist](https://phac-nml.github.io/irida-next/docs/development/development_processes/code_review#acceptance-checklist) for this PR.

### CODE REVIEW

**Review by JeffreyThiessen — APPROVED** [review]
Looks to be working as intended. :+1:

**Review by ChrisHuynh333 — APPROVED** [review]
Works as described!

File: test/graphql/update_sample_metadata_test.rb [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+                                 variables: { sampleId: sample.to_global_id.to_s,
+                                              metadata: { key1: 'value1' } })
+
+    assert_not_nil result['errors'], 'shouldn\'t work and have errors.'
+
+    error_message = result['errors'][0]['message']
+
+    assert_equal 'You are not authorized to perform this action', error_message
+  end
+end
```
Comment (deepsidhu85) [inline test/graphql/update_sample_metadata_test.rb:86]: Can you also add a test for attempting to update a sample that the user doesn't have permission to update

**Review by deepsidhu85 — CHANGES_REQUESTED** [review]
This looks great and is working as expected. Just one comment about adding a test below

File: test/graphql/update_sample_metadata_test.rb (hunk 1 above)
Comment (ksierks) [reply test/graphql/update_sample_metadata_test.rb:97]: 6ddabb475fb4656f331dbd9e58ad111ffb93fae5

**Review by deepsidhu85 — APPROVED** [review]
Looks good to me and working as described

**Pull request merged**